src/frontend/src/handler/transaction.rs

```diff
@@ -42,29 +43,42 @@ pub async fn handle_begin(
                 TransactionMode::AccessMode(mode) => {
                     let _ = access_mode.replace(mode);
                 }
-                TransactionMode::IsolationLevel(_) => not_impl!("ISOLATION LEVEL"),
+                TransactionMode::IsolationLevel(_) => {
+                    // Note: This is for compatibility with some external drivers (like postgres_fdw) that
+                    // always start a transaction with an Isolation Level.
+                    const MESSAGE: &str = "\
+                    Transaction with given Isolation Level is not supported yet.\n\
+                    For compatibility, this statement will still proceed with RepeatableRead.";
+                    notices.push(MESSAGE);
+                }
             }
         }
 
         match access_mode {
             Some(TransactionAccessMode::ReadOnly) => AccessMode::ReadOnly,
             Some(TransactionAccessMode::ReadWrite) | None => {
                 // Note: This is for compatibility with some external drivers (like psycopg2) that
-                // issue `BEGIN` implicitly for users. Not actually starting a transaction is okay
-                // since `COMMIT` and `ROLLBACK` are no-ops (except for warnings) when there is no
-                // active transaction.
+                // issue `BEGIN` implicitly for users.
                 const MESSAGE: &str = "\
                     Read-write transaction is not supported yet. Please specify `READ ONLY` to start a read-only transaction.\n\
-                    For compatibility, this statement will still succeed but no transaction is actually started.";
```

Comment:
This is interesting. Does the original compatibility manner (i.e., not actually starting a txn) work for `fdw`?

Reply:
For fdw, it should work. But cursors should only work in transaction. If we keep the original compatibility manner we must allow declare/fetch/close cursors out of transaction blocks. This is unexpected and risky. So I am inclined to change the compatibility manner.